### PR TITLE
Update sync daemonset annotation checks

### PR DESCRIPTION
**Pull request by vrutkovs**
* Use correct variable when checkin for expected number of master nodes
* Check that sync annotations were set on all nodes after schedulability is set

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1628485

### CODE REVIEW

File: roles/openshift_node_group/tasks/sync.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@
   retries: 60
   delay: 10
 
-- name: Wait for sync DS to set annotations on all nodes
+- name: Wait for sync DS to set annotations on master nodes
```
Comment (vrutkovs) [inline roles/openshift_node_group/tasks/sync.yml]: At this point only masters have been bootstrapped, so this check won't resolve 'missing infra nodes' issue

File: roles/openshift_node_group/tasks/sync.yml [hunk 2]
```diff
@@ -83,8 +83,9 @@
     - node_status.results.results is defined
     - node_status.results.results | length > 0
     - node_status.results.results[0]['items']
-     | map(attribute='metadata.annotations') | map('list') | flatten
-     | select('match', '[\"node.openshift.io/md5sum\"]') | list | length == node_status.results.results | length
```
Comment (vrutkovs) [inline roles/openshift_node_group/tasks/sync.yml]: `node_status.results.results | length` would always be 1, as `oc_obj` returns a single List object.

File: roles/openshift_manage_node/tasks/config.yml [hunk 3]
```diff
@@ -9,3 +9,21 @@
   until: node_schedulable is succeeded
   when: "'nodename' in openshift.node"
   delegate_to: "{{ openshift_master_host }}"
+
+- name: Wait for sync DS to set annotations on all nodes
```
Comment (vrutkovs) [inline roles/openshift_manage_node/tasks/config.yml]: Run sync DS check again after all node CSRs have been approved and node configs have landed

**Comment by mtnbikenc** [comment]
Hit this same error.  Testing this PR.

**Pull request merged**